Installation on Linux fails due to ignored write() result

I'm cloning the git repository to install powerline (per the instructions [here](https://powerline.readthedocs.org/en/latest/installation/linux.html#installation-on-linux)). I get the following error output when I run `pip install` (EDIT: I meant to say _warning_ here, not _error_):

```
Downloading/unpacking git+git://github.com/Lokaltog/powerline
  Cloning git://github.com/Lokaltog/powerline to /tmp/pip-aPTBJK-build
  Running setup.py egg_info for package from git+git://github.com/Lokaltog/powerline
    client/powerline.c: In function ‘main’:
    client/powerline.c:123:4: warning: ignoring return value of ‘write’, declared with attribute warn_unused_result [-Wunused-result]
        (void) write(STDOUT_FILENO, buf, (size_t) i);
        ^

Cleaning up...
```

It seems this may be related to gcc stripping the void cast as it processes `powerline.c` - [see this bug report](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509).

It looks like this would simple enough to fix; I added an unused variable and this seems to allow installation to work [seamlessly](https://github.com/ejstronge/powerline/tree/no_wur).

Is this a problem on my end? Has anyone else had issues with installations from the git repository?

--- a/client/powerline.c
+++ b/client/powerline.c
@@ -123,30 +123,7 @@ int main(int argc, char *argv[]) {
 			close(sd);
 			HANDLE_ERROR("read() failed");
 		} else if (read_size > 0) {
-			ptrdiff_t written = 0;
-			const char *bufp = &(buf[0]);
-			while (written < read_size) {
-				written = write(STDOUT_FILENO, buf, (size_t) read_size);
-				if (written >= 0) {
-					bufp += written;
-					read_size -= written;
-					written = 0;
-				} else {
-					errno = 0;
-					switch (errno) {
-						case EAGAIN:
-#if EAGAIN != EWOULDBLOCK
-						case EWOULDBLOCK:
-#endif
-						case EINTR:
-							break;
-						default:
-							(void) printf("Error while writing output: %s", strerror(errno));
-							close(sd);
-							return 2;
-					}
-				}
-			}
+			do_write(STDOUT_FILENO, buf, (size_t) read_size);
 		}
 	}
 

